compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1561,9 +1563,12 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
             let mut map = FxHashMap::with_capacity_and_hasher(end_id as usize, Default::default());
             for i in 0..end_id {
```

Comment:
Late to the party... but if there are gaps, rather than stopping at `end_id`, do we need to keep iterating until we've processed `self.root.tables.def_path_hashes.size()` elements?

(Or maybe something more efficient. I'm not familiar with how these tables work.)

Reply:
The `size()` method on `Lazy<Table>` has a somewhat misleading name - it returns the maximum possible value. So, iterating up to `end_id` guarnatees that we visit all present entries.